src/jaegertracing/TracerTest.cpp

```diff
@@ -463,6 +463,73 @@ TEST(Tracer, testPropagation)
     tracer->Close();
 }
 
+TEST(Tracer, testPropagationWithW3CHeaderAndFormat)
+{
+    const auto handle =
+        testutils::TracerUtil::installGlobalTracer(propagation::Format::W3C);
+    const auto tracer =
+        std::static_pointer_cast<Tracer>(opentracing::Tracer::Global());
+    const std::unique_ptr<Span> span(static_cast<Span*>(
+        tracer->StartSpanWithOptions("test-inject", {}).release()));
+    const auto spanContext = span->context();
+
+    std::ostringstream oss;
+    oss << "00";
+    oss << '-';
+    oss << std::setw(16) << std::setfill('0') << std::hex
```

Comment:
@tobiasstadler Can you help me understand why there are 0 in the first half of the trace id? Is there limitations of Jaeger?

Reply:
Nevermind, I found PR #254

Reply:
Yes, the jaeger client only generates 64bit trace ids. See #254.